server/src/main/java/org/opensearch/search/SearchModule.java

```diff
@@ -1290,7 +1291,17 @@ public FetchPhase getFetchPhase() {
     }
 
     public QueryPhase getQueryPhase() {
-        return (queryPhaseSearcher == null) ? new QueryPhase() : new QueryPhase(queryPhaseSearcher);
+        QueryPhase queryPhase;
+        if (queryPhaseSearcher == null) {
+            // use the defaults
+            queryPhase = new QueryPhase();
+        } else if (queryPhaseSearcher instanceof ConcurrentQueryPhaseSearcher) {
```

Comment:
Since the queryPhaseSearcher and aggregation processor are bundled together now, I think we should make this part of the context:

```
public interface QueryPhaseSearcher {
    default AggregationProcessor newAggregationProcessor() {
        return new DefaultAggregationProcessor();
    }
}
```

The `QueryPhase` should be able to configure itself properly all the time, no typecast should be required.